main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+var isTesting = false
+
+func main() {
+	inputChan := make(chan string)
+
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	file, err := os.Create("result.txt")
+	if err != nil {
+		log.Println("Ошибка создания файла:", err)
+		return
+	}
+	defer file.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go readInput(ctx, inputChan, os.Stdin, &wg)
+
+	wg.Add(1)
+	go writeToFile(ctx, inputChan, file, &wg)
+
+	go func() {
+		<-sigChan
+		log.Println("Получен сигнал завершения, завершаем работу")
+		cancel()
+	}()
+
+	<-ctx.Done()
+
+	wg.Wait()
```

Comment:
в целом, здесь немного оверхед получился, сигнал прерывания можно было бы ловить и в мейне
```suggestion

	<-sigChan
	log.Println("Получен сигнал завершения, завершаем работу")
	cancel()

	wg.Wait()
```

Reply:
понял-принял, спасибо